core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java

```diff
@@ -801,17 +809,49 @@ private <T> void validateAlterProperties(
       T... changes) {
     doWithCatalog(
         getCatalogIdentifier(ident),
-        c ->
-            c.doWithPropertiesMeta(
-                p -> {
-                  Map<String, String> upserts = getPropertiesForSet(changes);
-                  Map<String, String> deletes = getPropertiesForDelete(changes);
-                  validatePropertyForAlter(provider.apply(p), upserts, deletes);
-                  return null;
-                }),
+        c -> {
+          c.doWithPropertiesMeta(
+              p -> {
+                Map<String, String> upserts = getPropertiesForSet(changes);
+                Map<String, String> deletes = getPropertiesForDelete(changes);
+                validatePropertyForAlter(provider.apply(p), upserts, deletes);
+                return null;
+              });
+          c.doWithCapabilities(
+              caps -> {
+                caps.columnNotNull().apply(getColumnWithNullable(changes));
+                return null;
+              });
+          return null;
```

Comment:
If we handle caseSensitiveOnName, the code should like this:
```java
Column[] standardColumns =
        doWithCatalog(
            catalogIdent,
            c -> {
              // ......
              return c.doWithCapabilities(
                  caps -> {
                    caps.columnNotNull().apply(columns);
                    return caps.caseSensitiveOnName(Capability.Scope.COLUMN).apply(columns);
                  });
            },
            IllegalArgumentException.class);
// use `standardColumns` instead of `columns` to pass down
```
The `HasCapabilities` is like `CatalogCapabilities` and `CapabilityManger`; the `Capability<T, R, E extends Exception>` is like `CapabilityRule`

Reply:
This part of the code feels inelegant; it's procedural. You might consider optimizing it.

Reply:
Can you provide some code samples to show your idea?